### PR TITLE
Account for all layers in reset limits/zoom home

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request overrides the behavior in glue for `viewer.state.reset_limits()` (and therefore the zoom "home" button) to account for the limits of all *visible* layers.

**Previous behavior**:

https://user-images.githubusercontent.com/877591/206013547-a52a5c67-307d-40ac-b1da-e12d00aa03c7.mov

https://user-images.githubusercontent.com/877591/206261204-128c8dd7-bba3-48a3-a3ae-55e7457a8d46.mov

**New behavior**:

https://user-images.githubusercontent.com/877591/206013382-817561e2-c1fc-40de-a90e-f0986ee7f0a5.mov

https://user-images.githubusercontent.com/877591/206261235-ba632f57-90a5-42ee-97f9-eb8f91d233c7.mov

**TODO**:
- [x] fix CI failures
- [x] add any missing test coverage/scenarios

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by kecnry** [comment]
Drats, the check for `reference_data.coords` now means existing tests don't reach the new code.  I'll work on writing some new tests to get test coverage back up before marking back as ready-for-review.

File: jdaviz/core/freezable_state.py
Comment (javerbukh) [inline jdaviz/core/freezable_state.py:29]: Would it make sense to add this upstream, or is it better this lives in jdaviz for now?

**Review by javerbukh — APPROVED** [review]
This has definitely been a headache, thanks for fixing it! It would be nice if this worked at data load too, but I can see that being a follow up effort (if other people even want it). Nice!

**Comment by kecnry** [comment]
> It would be nice if this worked at data load too, but I can see that being a follow up effort (if other people even want it)

That's a good idea.  I considered having the limits reset when changing linking type, but that might not always be desired behavior.  Data loading is a little more complicated since I think we would only want that behavior at the initial startup, and not on future adding/removing data from the viewer.  I suppose we could loop through all viewers and reset within the parser, or try to modify the default limits within glue?

**Review by duytnguyendtn — APPROVED** [review]
Tested in Imviz and Specviz, works as intended. I'm quite surprised it took us this long to get around to adding this; thanks for adding it!

**Comment by kecnry** [comment]
@pllim wanted to take a look at the logic before merge, so will hold off for now.  Thanks for the reviews/approvals!

**Comment by camipacifici** [comment]
THANK YOU!

File: jdaviz/core/freezable_state.py
Comment (kecnry) [reply jdaviz/core/freezable_state.py:29]: @astrofrog - just confirming that you'd prefer both the changes to the profile viewer and image viewer to be in jdaviz and not in glue - at least for now - correct?

**Comment by pllim** [comment]
I also saw this slight vertical offset on linked pan/zoom mode but it went away when I close the tray, so maybe it is unrelated. 🤷 

Screenshot 2022-12-13 113906

File: jdaviz/configs/imviz/helper.py
Comment (pllim) [reply jdaviz/configs/imviz/helper.py:626]: Other than this, the rest looks fine at a glance. I don't grok all the limits stuff in core but minimal changes to Imviz tests means I don't worry.

File: jdaviz/core/freezable_state.py
Comment (pllim) [reply jdaviz/core/freezable_state.py:29]: We can always move it upstream later. Maybe open follow up issue in glue so this doesn't get lost in PR comments.

File: jdaviz/configs/imviz/helper.py
Comment (kecnry) [reply jdaviz/configs/imviz/helper.py:626]: This is a... missing feature 😜  Working on adding support for new viewers to automatically adopt the correct value for `state.linked_by_wcs` so that their `reset_limits` works as expected based on the linking.

Comment (kecnry) [reply jdaviz/configs/imviz/helper.py:626]: the latest commit should handle this case (see recording).  Let me know if you like that solution or see any traps.

https://user-images.githubusercontent.com/877591/207409523-e69e694e-18d3-43c8-9e8c-a7ae87e4fb09.mov

**Comment by kecnry** [comment]
> I also saw this slight vertical offset on linked pan/zoom mode but it went away when I close the tray, so maybe it is unrelated. 

I am quite sure that is unrelated (this PR only affects clicking the home button or calling `reset_limits`)

File: jdaviz/app.py
Comment (pllim) [inline jdaviz/app.py:1667]: Did you close the parenthesis at the wrong place?
```suggestion
            viewer.state.linked_by_wcs = list(self._viewer_store.values())[0].state.linked_by_wcs
```

Comment (kecnry) [reply jdaviz/app.py:1667]: woops!  Good catch!

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**
